pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -29,13 +31,15 @@ const (
 // CollectorBundle is a container for a group of collectors. It provides a way
 // to easily run them all.
 type CollectorBundle struct {
-	check              *OrchestratorCheck
-	collectors         []collectors.Collector
-	discoverCollectors bool
-	extraSyncTimeout   time.Duration
-	inventory          *inventory.CollectorInventory
-	stopCh             chan struct{}
-	runCfg             *collectors.CollectorRunConfig
+	check               *OrchestratorCheck
+	collectors          []collectors.Collector
+	discoverCollectors  bool
+	extraSyncTimeout    time.Duration
+	inventory           *inventory.CollectorInventory
+	stopCh              chan struct{}
+	runCfg              *collectors.CollectorRunConfig
+	crdDiscovery        *crd.DiscoveryCollector
+	activatedCollectors map[string]struct{}
```

Comment:
this is to make sure we don't activate the same collector configured (crd or normal) resource multiple times

Reply:
This code path does it: https://github.com/DataDog/datadog-agent/blob/25c02455ab8743da315bcbfe62aa78a10376eca9/pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go#L123-L126

Reply:
ok gotcha